client/Components/Dogs/Dogs.jsx

```diff
@@ -2,30 +2,61 @@ import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
 
-import FilterSideBar from './../FilterSideBar';
+import FilterSideBar from '../FilterSideBar/FilterSideBar';
 import { ROUTE_PATH } from './../../commons/constants';
 const { DOGS } = ROUTE_PATH;
 
 class Dogs extends Component {
   constructor() {
     super();
     this.state = {
-      breedId: 0
+      breedId: 0,
+      gender: '',
+      size: ''
     };
   }
 
-  updateBreed = breedId => this.setState({ breedId });
+  updateFilter = filterObject => this.setState(filterObject);
 
-  render() {
-    const { breedId } = this.state;
+  filter = (sourceList, criteria, filterFunction) => {
+    return criteria ? sourceList : sourceList.filter(filterFunction);
+  };
+
+  filterDogs = () => {
```

Comment:
@inKEYYgnito  i'll retain this code for now because the filtering logic gets complicated if i put them all in 1 filter. we can work on optimizing this later on.